src/test/scala/com/fulcrumgenomics/util/NumericCounterTest.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.util
+
+import com.fulcrumgenomics.testing.UnitSpec
+
+/**
+  * Tests for NumericCounter.
+  */
+class NumericCounterTest extends UnitSpec {
+
+  "NumericCounter" should "operate on Ints" in {
+    val counter = new NumericCounter[Int]()
+    Stream.range(1, 10, 1).foreach { i => counter.count(i, 1) }
+    counter.count(1)
+    counter.mean() shouldBe 4.6 +- 0.0001
+    counter.standardDeviation() shouldBe  2.875181 +- 0.0001
+    counter.mode() shouldBe 1
+    counter.median() shouldBe 4
+    counter.medianAsDouble() shouldBe 4.5
+  }
+
+  it should "operate on Doubles" in {
+    val counter = new NumericCounter[Double]()
+    Stream.range(1, 10, 1).foreach { i => counter.count(i / 10.0, 1) }
+    counter.count(1 / 10.0)
+    counter.mean() shouldBe 0.46 +- 0.0001
+    counter.standardDeviation() shouldBe  0.2875181 +- 0.0001
+    counter.mode() shouldBe 0.1
+    counter.median() shouldBe 0.45
+    counter.medianAsDouble() shouldBe 0.45
+  }
+}
```

Comment:
Could you please add tests to make sure that the methods all do the "right thing" on counters that have:
  - zero counts added
  - a single count added
  - two counts added